hypershift-operator/controllers/scheduler/autoscaler.go

```diff
@@ -653,3 +728,121 @@ func podNames(pods []corev1.Pod) []string {
 	}
 	return names
 }
+
+func validateConfigForNonRequestServing(config *schedulingv1alpha1.ClusterSizingConfiguration) error {
+	if condition := meta.FindStatusCondition(config.Status.Conditions, schedulingv1alpha1.ClusterSizingConfigurationValidType); condition == nil || condition.Status != metav1.ConditionTrue {
+		msg := ""
+		if condition != nil {
+			msg = condition.Message
+		}
+		return fmt.Errorf("cluster sizing configuration is not valid: %s", msg)
+	}
+
+	for _, sizeConfig := range config.Spec.Sizes {
```

Comment:
Why do we require that every size have this configured for the controller to take any action?

Reply:
Because we scale the non-request serving nodes to exactly the weighted sum of all hotedclusters. If you don't have a value configured for a particular size we are ignoring hosted clusters at that size, and that can be bad.

Reply:
I see, that makes sense. Could you express this in CEL or add it to the validation controller and expose with the valid config status?

Reply:
added to the validation